project-management-infrastructure/src/main/java/life/qbic/projectmanagement/infrastructure/DataManagerVault.java

```diff
@@ -0,0 +1,198 @@
+import java.util.Optional;
+import javax.crypto.spec.SecretKeySpec;
+import life.qbic.logging.api.Logger;
+import org.springframework.beans.factory.DisposableBean;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Component;
+
+/**
+ * Container for a Java keystore to maintain secrets within the application.
+ *
+ * @since 1.8.0
+ */
+@Component
+public class DataManagerVault implements DisposableBean {
+
+  public static final String UNEXPECTED_VAULT_EXCEPTION = "Unexpected vault exception";
+  private static final Logger log = logger(DataManagerVault.class);
+  private static final String KEY_GENERATOR_ALGORITHM = "AES";
+  private static final double MIN_ENTROPY = 100; // Shannon entropy * length of secret
+  private final KeyStore keyStore;
+  private final String envVarKeystorePassword;
+  private final String envVarKeystoreEntryPassword;
+  private final Path keystorePath;
+
+  public DataManagerVault(@Value("${qbic.security.vault.key.env}") String vaultKeyEnvVar,
+      @Value("${qbic.security.vault.path}") String vaultPathString,
+      @Value("${qbic.security.vault.entry.password.env}") String vaultEntryPassword)
+      throws CertificateException, KeyStoreException, IOException, NoSuchAlgorithmException {
+    if (System.getenv(vaultKeyEnvVar) == null) {
+      throw new DataManagerVaultException(
+          "Cannot find value for environment variable: %s".formatted(vaultKeyEnvVar));
+    }
+    if (System.getenv(vaultEntryPassword) == null) {
+      throw new DataManagerVaultException(
+          "Cannot find value for environment variable: %s".formatted(vaultEntryPassword)
+      );
+    }
+    this.envVarKeystoreEntryPassword = vaultEntryPassword;
+    this.envVarKeystorePassword = vaultKeyEnvVar;
+
+    double entropy;
+    if ((entropy = calculateEntropy(System.getenv(envVarKeystorePassword))) < MIN_ENTROPY) {
+      throw new DataManagerVaultException(
+          "Entry of password for keystore was to low: %f (min: %f)".formatted(entropy,
+              MIN_ENTROPY));
+    }
+    if ((entropy = calculateEntropy(System.getenv(envVarKeystoreEntryPassword))) < MIN_ENTROPY) {
+      throw new DataManagerVaultException(
+          "Entry of password for keystore entries was to low: %f (min: %f)".formatted(entropy,
+              MIN_ENTROPY));
+    }
+
+    this.keystorePath = fromString(vaultPathString);
+    this.keyStore = createVault(vaultKeyEnvVar, keystorePath);
+  }
+
+  // Calculates the product of Shannon entropy and secret length
+  // See https://en.wikipedia.org/wiki/Entropy_(information_theory)
+  private static double calculateEntropy(String secret) {
+    if (secret == null || secret.isEmpty()) {
+      return 0.0;
+    }
+
+    Map<Character, Integer> frequencyMap = new HashMap<>();
+    int length = secret.length();
+
+    // Count character frequencies
+    for (char c : secret.toCharArray()) {
+      frequencyMap.put(c, frequencyMap.getOrDefault(c, 0) + 1);
```

Comment:
NIT: Think about extracting a variable for the current count here. This way accessing and modifying the hashmap is more separated in the code.